File: packages/js/src/zipToCoords.ts
import rawData from "./data/data_dict.json" with { type: "json" };
const data = rawData as unknown as Record<string, [number, number]>;


export default function zipToCoords(
    zip: string
): { lat: number | null, lon: number | null}
{
    const regex = /^\d{5}$/;
    if (!regex.test(zip)) {
        return { lat: null, lon: null};
    }
    const coords = data[zip];
    if (!coords) {
        return { lat: null, lon: null };
    }
    return { lat: coords[0], lon: coords[1] };
}

File: packages/py/zipcoords/zip_to_coords.py
import json
from importlib.resources import files


def zip_to_coords(zip: str) -> tuple[float, float]:
    if (
        not isinstance(zip, str)
        or not zip.isdigit()
        or len(zip) != 5
    ):
        return None, None

    path = files("zipcoords.data").joinpath("data_dict.json")
    with open(path, 'r') as f:
        data: dict[str, list[float, float]] = json.load(f)

    coords = data.get(zip)
    if not coords:
        return None, None

    return coords[0], coords[1]
